examples/stm32/f0/other/can/can.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2010 Thomas Otto <tommi@viadmin.org>
+ * Copyright (C) 2010 Piotr Esden-Tempski <piotr@esden.net>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/cm3/systick.h>
+#include <libopencm3/stm32/can.h>
+
+
+#define TS1FROMBTRVAL(value)	(value & 0xF0000) //((1<<16)|(1<<17)(1<<18)(1<<19)))
+#define TS2FROMBTRVAL(value)	(value & 0x700000)//((1<<20)|(1<<21)(1<<22)))
+#define SJWFROMBTRVAL(value)	(value & 0x3000000)//((1<<24)|(1<<24)))
+#define BRPFROMBTRVAL(value)	((value & 0x1FF)+1)
+
+ /* 500kbit/s -> CAN_BTR = 0x001c0005 http://www.bittiming.can-wiki.info/ */
+#define BTR500k 0x001c0005
+
+struct can_tx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+};
+
+struct can_rx_msg {
+	uint32_t std_id;
+	uint32_t ext_id;
+	uint8_t ide;
+	uint8_t rtr;
+	uint8_t dlc;
+	uint8_t data[8];
+	uint8_t fmi;
+};
+
+struct can_tx_msg can_tx_msg;
+struct can_rx_msg can_rx_msg;
+
+
+static void systick_setup(void)
+{
+	/* 72MHz / 8 => 9000000 counts per second */
+	systick_set_clocksource(STK_CSR_CLKSOURCE_AHB);
+
+	/* 9000000/9000 = 1000 overflows per second - every 1ms one interrupt */
+	/* SysTick interrupt every N clock pulses: set reload to N-1 */
+	systick_set_reload(8999);
+
+	systick_interrupt_enable();
+
+	/* Start counting. */
+	systick_counter_enable();
+}
+
+static void can_setup(void)
+{
+	/* Enable peripheral clocks. */
+
+	//B8 Rx
+	//B9 Tx
```

Comment:
Stylecheck may be the issue now. 

C99 introduced the notion that both // and /* ... */ were legal comments in **C code** (adopting the C++ convention that existed at the time.) Now that LOC3 is no longer restricted to being C89 compliant (the latest commits have started using C99 features  such as variable declaration in the loop context).  Perhaps it is time to update `make stylecheck` to only enforce C99 ?